test/kitchen/test/integration/common/rspec/spec_helper.rb

```diff
@@ -0,0 +1,417 @@
+    else
+      conf_path = "#{ENV['ProgramData']}\\Datadog\\datadog.yaml"
+      open(conf_path, 'w') do |f|
+        f.puts "apm_enabled: false"
+      end
+    end
+    output = restart
+    if os != :windows
+      expect(output).to be_truthy
+      system 'command -v systemctl 2>&1 > /dev/null || sleep 5 || true'
+    else
+      sleep 5
+    end
+    expect(is_port_bound(8126)).to be_falsey
+  end
+
+  it "doesn't say 'not running' in the info command" do
+    expect(info).to_not include 'not running'
+  end
+end
+
+shared_examples_for 'an Agent that stops' do
+  it 'stops' do
+    output = stop
+    if os != :windows
+      expect(output).to be_truthy
+    end
+    expect(is_running?).to be_falsey
+  end
+
+  it 'has connection refuse in the info command' do
+    if os == :windows
+      expect(info).to include 'No connection could be made'
+    else
+      expect(info).to include 'connection refuse'
+    end
+  end
+
+  it 'is not running any agent processes' do
+    sleep 5 # need to wait for the Agent to stop
+    expect(agent_processes_running?).to be_falsey
+  end
+
+  it 'starts after being stopped' do
+    output = start
+    if os != :windows
+      expect(output).to be_truthy
+    end
+    expect(status).to be_truthy
+  end
+end
+
+shared_examples_for 'an Agent that restarts' do
+  it 'restarts when the agent is running' do
+    if !is_running?
+      start
+    end
+    output = restart
+    if os != :windows
+      expect(output).to be_truthy
+    end
+    expect(is_running?).to be_truthy
+  end
+
+  it 'restarts when the agent is not running' do
+    if is_running?
+      stop
+    end
+    output = restart
+    if os != :windows
+      expect(output).to be_truthy
+    end
+    expect(status).to be_truthy
+  end
+end
+
+shared_examples_for 'an Agent that is removed' do
+  it 'should remove the agent' do
+    if os == :windows
+      # uninstallcmd = "start /wait msiexec /q /x 'C:\\Users\\azure\\AppData\\Local\\Temp\\kitchen\\cache\\ddagent-cli.msi'"
+      uninstallcmd='for /f "usebackq" %n IN (`wmic product where "name like \'datadog%\'" get IdentifyingNumber ^| find "{"`) do start /wait msiexec /log c:\\uninst.log /q /x %n'
+      expect(system(uninstallcmd)).to be_truthy
+    else
+      if system('which apt-get &> /dev/null')
+        expect(system("sudo apt-get -q -y remove datadog-agent > /dev/null")).to be_truthy
+      elsif system('which yum &> /dev/null')
+        expect(system("sudo yum -y remove datadog-agent > /dev/null")).to be_truthy
+      elsif system('which zypper &> /dev/null')
+        expect(system("sudo zypper --non-interactive remove datadog-agent > /dev/null")).to be_truthy
+      else
+        raise 'Unknown package manager'
+      end
+    end
+  end
+
+  it 'should not be running the agent after removal' do
+    sleep 5
+    expect(agent_processes_running?).to be_falsey
+  end
+
+  it 'should remove the agent binary' do
+    if os != :windows
```

Comment:
can you implement that test for windows too? :) (unless I've missed something and the test is more complex than just checking for the existence of `C:\\Program Files\\Datadog\\Datadog Agent\\embedded\\agent.exe`)

Reply:
I meant to do this and must have forgotten, thank you!